### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by SamChou19815**
### Summary 

Bump dependencies and regenerate `yarn.lock`. Hopefully it can help solve some yarn issues.

### Test Plan 

CI. Every preview site should still be fine.

### CODE REVIEW

**Comment by SamChou19815** [comment]
#173 is fine, which probably means that some dependency really mess up in this one...

File: backend/package.json [hunk 1]
```diff
@@ -4,38 +4,35 @@
   "description": "The backend for the DTI Internal DTI Organization Logic",
   "main": "index.js",
   "scripts": {
-    "start": "./node_modules/.bin/netlify-lambda serve build",
-    "dev": "ts-node --transpile-only src/api.ts",
+    "start": "netlify-lambda serve build",
+    "dev": "esr src/api.ts",
```
Comment (SamChou19815) [inline backend/package.json:8]: Use esbuild runner to run it much faster than `ts-node`

File: netlify.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[functions]
+  node_bundler = "esbuild"
```
Comment (SamChou19815) [inline netlify.toml:2]: Switch the function bundler to esbuild. See https://www.netlify.com/blog/2021/04/02/modern-faster-netlify-functions/. The default one seems to have some bugs on dynamic requires.

Comment (henry-li-06) [reply netlify.toml:2]: Is this what was causing the build issues for the netlify functions?

Comment (SamChou19815) [reply netlify.toml:2]: Yeah I believe. I look into the files that failed and it contains a dynamic require.

**Review by henry-li-06 — APPROVED** [review]
LGTM

**Pull request merged**